training/vision/components/image_classification/spec.yaml

```diff
@@ -5,72 +5,168 @@ description: Trains an image classification model. [Learn More](https://aka.ms/b
+  learning_rate_scheduler:
+    description: Type of learning rate scheduler. Must be warmup_cosine or step.
+    type: string
+    optional: true
+    default: warmup_cosine
+    enum: ['warmup_cosine', 'step']
   model_name:
     type: string
     description: Model name
     default: vitb16r224
-    enum: ['mobilenetv2', 'resnet18', 'resnet34', 'resnet50', 'resnet101', 'resnet152', 'resnest50', 'resnest101', 'seresnext', 'vits16r224', 'vitb16r224', 'vitl16r224']
-  learning_rate:
-    description: Initial learning rate
+  momentum:
+    description: Value of momentum when optimizer is sgd. Must be a float in the range [0, 1].
     type: number
     optional: true
     min: 0
     max: 1
+  nesterov:
+    description: Enable nesterov when optimizer is sgd.
+    type: boolean
+    optional: true
   number_of_epochs:
     description: Number of training epochs
     type: integer
-    default: 15
+    optional: true
     min: 1
   optimizer:
     description: Type of optimizer
     type: string
+    optional: true
     default: sgd
     enum: ['sgd', 'adam', 'adamw']
+  step_lr_gamma:
+    description: Value of gamma when learning rate scheduler is step. Please check for https://learn.microsoft.com/azure/machine-learning/reference-automl-images-hyperparameters more information.
+    type: number
+    optional: true
+    min: 0
+    max: 1
+  step_lr_step_size:
+    description: Value of step size when learning rate scheduler is step. Please check for https://learn.microsoft.com/azure/machine-learning/reference-automl-images-hyperparameters more information.
+    type: integer
+    optional: true
+    min: 0
   training_batch_size:
-    description: Training batch size
+    description: Training batch size.
+    type: integer
+    optional: true
+    min: 1
+  training_crop_size:
+    description: Image crop size that's input to your neural network for training dataset. Notes - seresnext doesn't take an arbitrary size. ViT-variants should have the same validation_crop_size and training_crop_size.
     type: integer
     optional: true
     min: 1
   validation_batch_size:
-    description: Validation batch size
+    description: Validation batch size.
+    type: integer
+    optional: true
+    min: 1
+  validation_crop_size:
+    description: Image crop size that's input to your neural network for validation dataset. Note - seresnext doesn't take an arbitrary size. ViT-variants should have the same validation_crop_size and training_crop_size.
     type: integer
     optional: true
     min: 1
+  validation_resize_size:
+    description: Image size to which to resize before cropping for validation dataset. Note - seresnext doesn't take an arbitrary size.
+    type: integer
+    optional: true
+    min: 1
+  warmup_cosine_lr_cycles:
+    description: Value of cosine cycle when learning rate scheduler is warmup_cosine. Please check for https://learn.microsoft.com/azure/machine-learning/reference-automl-images-hyperparameters more information.
```

Comment:
nit: wonder if URL needs to be or could be hyperlinked in description? not a big deal, just thinking out loud!

Reply:
synced offline..